TESTS/mbed_hal/device_key/main.cpp

```diff
@@ -0,0 +1,126 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utest/utest.h"
+#include "unity/unity.h"
+#include "greentea-client/test_env.h"
+#include "mbed.h"
+#include "device_key_api.h"
+
+#ifndef DEVICE_DEVKEY
+#error [NOT_SUPPORTED] DEVICE_KEY needs to be enabled for this test
+#endif
+
+using namespace utest::v1;
+
+enum TestDeviceKeySize {
+
+    DEVICE_KEY_16BYTE = 16,
+    DEVICE_KEY_32BYTE = 32
+};
+
```

Comment:
Why is this enum? Wouldn't a define or a const be better if you care about the values?